WcaOnRails/spec/requests/registrations_spec.rb

```diff
@@ -187,6 +181,96 @@
+            ["a", user.name, user.country.name, "", user.dob, user.gender, user.email, "1", "0"],
+          ]
+          expect {
+            post competition_registrations_do_import_path(competition), params: { registrations_import: { registrations_file: file } }
+          }.to not_change { competition.registrations.count }
+            .and not_change { registration.reload.accepted_at }
+        end
+      end
+
+      context "CSV registrant already accepted in the database, but with different events" do
+        it "only updates registration events" do
+          registration = FactoryBot.create(:registration, :accepted, competition: competition, events: %(333))
+          user = registration.user
+          file = csv_file [
+            ["Status", "Name", "Country", "WCA ID", "Birth date", "Gender", "Email", "333", "444"],
+            ["a", user.name, user.country.name, "", user.dob, user.gender, user.email, "1", "1"],
+          ]
+          expect {
+            post competition_registrations_do_import_path(competition), params: { registrations_import: { registrations_file: file } }
+          }.to not_change { competition.registrations.count }
+            .and not_change { registration.reload.accepted_at }
+            .and change { registration.reload.events.map(&:id) }.from(%w(333)).to(%w(333 444))
+        end
+      end
+
+      context "CSV registrant already in the database, but deleted" do
+        it "acceptes the registration again" do
+          registration = FactoryBot.create(:registration, :deleted, competition: competition)
+          user = registration.user
+          file = csv_file [
+            ["Status", "Name", "Country", "WCA ID", "Birth date", "Gender", "Email", "333", "444"],
+            ["a", user.name, user.country.name, "", user.dob, user.gender, user.email, "1", "0"],
+          ]
+          expect {
+            post competition_registrations_do_import_path(competition), params: { registrations_import: { registrations_file: file } }
+          }.to not_change { competition.registrations.count }
+            .and change { registration.reload.accepted_at }
+          expect(registration.reload).to be_accepted
+        end
+      end
+
+      context "registrant deleted in the database, but not in the CSV file" do
+        it "leaves the registration unchanged" do
+          registration = FactoryBot.create(:registration, :deleted, competition: competition)
+          file = csv_file [
+            ["Status", "Name", "Country", "WCA ID", "Birth date", "Gender", "Email", "333", "444"],
+          ]
+          expect {
+            post competition_registrations_do_import_path(competition), params: { registrations_import: { registrations_file: file } }
+          }.to not_change { competition.registrations.count }
+            .and not_change { registration.reload.deleted_at }
+          expect(registration.reload).to be_deleted
```

Comment:
I'm curious about the reasoning behind this decision. I actually think it would make sense to un-delete the registration if the CSV has it.

Reply:
Sure thing! It's just that this scenario is `registrant deleted in the database, but *not* in the CSV file`.
The scenario you have in mind is the one above and I believe its's handled just as you'd expect =)